src/Symfony/Bridge/Twig/CHANGELOG.md

```diff
@@ -5,6 +5,7 @@ CHANGELOG
 ---
 
  * Deprecate passing a tag to the constructor of `FormThemeNode`
+ * Add the `Twig` constraint for validating Twig content
```

Comment:
I'd like to understand what is "valid Twig content".  

(version ? runtime ? context ?)

And i'm not sure any other validator return results that changing from one app to another.

Reply:
> And i'm not sure any other validator return results that changing from one app to another.

`ExpressionSyntaxValidator` behaves in a similar way, I suppose.

Reply:
ExpressionSyntax has flags for unknown functions / variables... but it's not equivalent as i see it.

Here, you can use variable not available in the context it would  be "valid". 

✅ Valid 

```
{% set division = 1000 / (non_existing_var ?? 0) %}

{{ dump(app.request) }}

{{ include('templates/admin/user_settings.html.twig') }}

{{ impersonation_path('admin@example.com') }}

```

❌ Invalid if the packages are not installed 

```

{{ 2024|serialize }}

{{ csrf_token('dddf') }}

{{ 'sdsdf'|sanitize_html) }}

```

I do believe this would require a giant red warning on the documentation page to explain the risks here.

Reply:
> I'd like to understand what is "valid Twig content".
> 
> (version ? runtime ? context ?)

It validates a value intended to be a Twig template to avoid syntax runtime errors. It behaves exactly like `twig:lint` command.

Reply:
I know. Only a developer can use the command.. i'm just afraid of what people could do if not informed enough of the risks.  But...  it's probably just me and my obsessions ;)